File: frontend/src/types/sweet.ts
export interface Sweet {
  id: number;
  name: string;
  description?: string;
  price: number;
  quantity: number;
  category?: string;
  image_url?: string;
  created_at: string;
  updated_at: string;
}

export interface SweetCreate {
  name: string;
  description?: string;
  price: number;
  quantity?: number;
  category?: string;
  image_url?: string;
}

export interface SweetUpdate {
  name?: string;
  description?: string;
  price?: number;
  quantity?: number;
  category?: string;
  image_url?: string;
}

export type CreateSweetRequest = SweetCreate;
export type UpdateSweetRequest = SweetUpdate;

export interface SearchSweetsParams {
  q?: string;
  category?: string;
  skip?: number;
  limit?: number;
}

File: backend/app/schemas/sweet.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class SweetBase(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    quantity: int = 0
    category: Optional[str] = None
    image_url: Optional[str] = None


class SweetCreate(SweetBase):
    pass


class SweetUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    price: Optional[float] = None
    quantity: Optional[int] = None
    category: Optional[str] = None
    image_url: Optional[str] = None


class Sweet(SweetBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
